src/test/run-make-fulldeps/c-link-to-rust-va-list-fn/checkrust.rs

```diff
@@ -16,127 +16,65 @@ extern crate libc;
 
 use libc::{c_char, c_double, c_int, c_long, c_longlong};
 use std::ffi::VaList;
-use std::slice;
-use std::ffi::CStr;
+use std::ffi::{CString, CStr};
 
-#[repr(C)]
-#[derive(Clone, Copy, Debug)]
-pub enum AnswerType {
-    Double,
-    Long,
-    LongLong,
-    Int,
-    Byte,
-    CStr,
-    Skip,
+macro_rules! continue_if {
+    ($cond:expr) => {
+        if !($cond) {
+            return 0xff;
+        }
+    }
 }
 
-#[repr(C)]
-pub union AnswerData {
-    pub double: c_double,
-    pub long: c_long,
-    pub longlong: c_longlong,
-    pub int: c_int,
-    pub byte: c_char,
-    pub cstr: *const c_char,
-    pub skip_ty: AnswerType,
+unsafe fn compare_c_str(ptr: *const c_char, val: &str) -> bool {
+    let cstr0 = CStr::from_ptr(ptr);
+    let cstr1 = CString::new(val).unwrap();
```

Comment:
If this fails, this would panic across an FFI boundary, which is UB. Not expected to happen, but doing something more conservative (return false / abort) would be better.

Reply:
FWIW this'll soon become a deterministic abort (not UB), so this is probably fine to leave in